lib/api/src/createReadStream.js

```diff
@@ -21,6 +21,14 @@ class SmbReadableStream extends Readable {
     }
     this.fileLength = fileLength
     this.wait = false
+
+    if (options) {
+      let { end } = options
+
+      if (end >= 0 && end < this.fileLength) {
+        this.fileLength = end + 1
+      }
+    }
```

Comment:
Start taking care at all options at the beginning (opportunity to document them all in one place if necessary)

`let {start, end} = options || {}` and use them later...